pkg/postgres/wal.go

```diff
@@ -193,3 +194,14 @@ func (segment Segment) NextSegments(size int, postgresVersion *int, segmentSize
 
 	return result
 }
+
+// BuildWALPath constructs the full destination path for WAL operations.
+// If walPath is already absolute, it returns it as-is.
+// If walPath is relative, it joins it with pgData.
+// This prevents path duplication when PostgreSQL or pg_rewind passes absolute paths.
+func BuildWALPath(pgData, walPath string) string {
+	if !filepath.IsAbs(walPath) {
+		return filepath.Join(pgData, walPath)
+	}
+	return walPath
+}
```

Comment:
Are we sure this is the right approach? It is pretty alarming that we don't know if we have a relative or absolute path in the input. I would prefer to standardize it so that it is always absolute, if possible.

Reply:
I dont think this Is the correct.place to raise this issue.

This Is a minimal fix and the function Is used to centralize the already existing logic. No ch anges were introduced.